diff --git a/node_modules/@walletconnect/jsonrpc-provider/dist/cjs/provider.js b/node_modules/@walletconnect/jsonrpc-provider/dist/cjs/provider.js
index 86c27ae..4dd3bb8 100644
--- a/node_modules/@walletconnect/jsonrpc-provider/dist/cjs/provider.js
+++ b/node_modules/@walletconnect/jsonrpc-provider/dist/cjs/provider.js
@@ -48,6 +48,15 @@ class JsonRpcProvider extends jsonrpc_utils_1.IJsonRpcProvider {
                         reject(e);
                     }
                 }
+                this.connection.on("payload", response => {
+                    if (response.id !== request.id) return;
+                    if (isJsonRpcError(response)) {
+                        reject(response.error.message);
+                    }
+                    else {
+                        resolve(response.result);
+                    }
+                });
                 this.events.on(`${request.id}`, response => {
                     if (jsonrpc_utils_1.isJsonRpcError(response)) {
                         reject(response.error.message);
diff --git a/node_modules/@walletconnect/jsonrpc-provider/dist/esm/provider.js b/node_modules/@walletconnect/jsonrpc-provider/dist/esm/provider.js
index 6271b24..f540878 100644
--- a/node_modules/@walletconnect/jsonrpc-provider/dist/esm/provider.js
+++ b/node_modules/@walletconnect/jsonrpc-provider/dist/esm/provider.js
@@ -37,6 +37,15 @@ export class JsonRpcProvider extends IJsonRpcProvider {
                     reject(e);
                 }
             }
+            this.connection.on("payload", response => {
+                if (response.id !== request.id) return;
+                if (isJsonRpcError(response)) {
+                    reject(response.error.message);
+                }
+                else {
+                    resolve(response.result);
+                }
+            });
             this.events.on(`${request.id}`, response => {
                 if (isJsonRpcError(response)) {
                     reject(response.error.message);
